src/intervals/arithmetic/power.jl

```diff
@@ -1,412 +1,440 @@
 
-    else # even power
-        if n > 0
-            if inf(a) ≥ 0
-                return @round(BigFloat, inf(a)^n, sup(a)^n)
-            elseif sup(a) ≤ 0
-                return @round(BigFloat, sup(a)^n, inf(a)^n)
-            else
-                return @round(BigFloat, mig(a)^n, mag(a)^n)
-            end
+# Examples
 
-        else
-            if inf(a) ≥ 0
-                return @round(BigFloat, sup(a)^n, inf(a)^n)
-            elseif sup(a) ≤ 0
-                return @round(BigFloat, inf(a)^n, sup(a)^n)
-            else
-                return @round(BigFloat, mag(a)^n, mig(a)^n)
-            end
-        end
-    end
-end
+```jldoctest
+julia> setdisplay(:full);
 
-"""
-    ^(x::BareInterval, y::BareInterval)
+julia> bareinterval(2, 3) ^ bareinterval(2)
+BareInterval{Float64}(4.0, 9.0)
 
-Implement the `pow` function of the IEEE Standard 1788-2015 (Table 9.1).
-"""
-^(x::BareInterval{T}, y::BareInterval{T}) where {T<:NumTypes} = BareInterval{T}(_bigequiv(x)^y)
+julia> interval(-1, 1) ^ interval(3)
+Interval{Float64}(-1.0, 1.0, com)
 
-^(x::BareInterval, y::BareInterval) = ^(promote(x, y)...)
+julia> interval(-1, 1) ^ interval(-3)
+Interval{Float64}(-Inf, Inf, trv)
+```
+"""
+function Base.:^(x::BareInterval, y::BareInterval)
+    isthininteger(y) && return _select_pown(power_mode(), x, Integer(sup(y)))
+    return _select_pow(power_mode(), x, y)
+end
 
-function ^(x::BareInterval{BigFloat}, y::BareInterval)
-    isempty_interval(y) && return y
-    domain = _unsafe_bareinterval(BigFloat, zero(BigFloat), typemax(BigFloat))
-    x = intersect_interval(x, domain)
-    isempty_interval(x) && return x
-    return hull(_pow(x, inf(y)), _pow(x, sup(y)))
+function Base.:^(x::Interval, y::Interval)
+    isthininteger(y) && return _select_pown(power_mode(), x, Integer(sup(y)))
+    return _select_pow(power_mode(), x, y)
 end
 
-# function _pow(a::BareInterval{T}, x::AbstractFloat) where {T<:Rational}
-#     a = unsafe_interval(float(T), inf(a).num/inf(a).den, sup(a).num/sup(a).den)
-#     return BareInterval{T}(a^x)
-# end
+Base.:^(n::Integer, y::Interval) = ^(n//one(n), y)
+Base.:^(x::Interval, n::Integer) = ^(x, n//one(n))
+Base.:^(x::Rational, y::Interval) = ^(convert(Interval{typeof(x)}, x), y)
+Base.:^(x::Interval, y::Rational) = ^(x, convert(Interval{typeof(y)}, y))
 
-_pow(a::BareInterval{BigFloat}, b::AbstractFloat) = _pow(a, big(b))
+# overwrite behaviour for small integer powers from https://github.com/JuliaLang/julia/pull/24240
+Base.literal_pow(::typeof(^), x::Interval, ::Val{n}) where {n} = x^n
 
-function _pow(a::BareInterval{BigFloat}, x::BigFloat)
-    domain = _unsafe_bareinterval(BigFloat, zero(BigFloat), typemax(BigFloat))
+# helper functions for power
 
-    if isthinzero(a)
-        x > 0 && return zero(BareInterval{BigFloat})
-        return emptyinterval(BareInterval{BigFloat})
-    end
+_select_pow(::PowerMode{:fast}, x, y) = fastpow(x, y)
+_select_pown(::PowerMode{:fast}, x, y) = fastpown(x, y)
```

Comment:
According to the docstrings (above), this should correspond to `pown(x, y)`, isn't it?

Reply:
Ah the docstring is wrong; I figured we should have a `fastpown` function as well. I will fix it.